File: Backend/PixelNestBackend/PixelNestBackend/Dto/PostDto.cs
﻿namespace PixelNestBackend.Dto
{
    public class PostDto
    {
        public string? PostDescription { get; set; }
        public string OwnerUsername { get; set; }
        public string PhotoDisplay { get; set; }
        public string? Location { get; set; }
        public List<IFormFile> Photos { get; set; } 
    }
}

File: Frontend/PixelNestFrontend/src/app/core/dto/post.dto.ts
import { CommentDto } from "./comment.dto";
import { ImagePathDto } from "./imagePath.dto";
import { LikedByUsers } from "./likedByUsers.dto";
import { SavedPosts } from "./savePost.dto";

export interface PostDto {
    postID: string;
    userID: string;
    ownerUsername: string;
    postDescription: string;
    totalComments: number;
    isDeletable:boolean;
    totalLikes: number;
    likedByUsers:LikedByUsers[];
    allComments:CommentDto[];
    imagePaths:ImagePathDto[];
    publishDate:Date;
    savedByUsers:SavedPosts[];
    location:string;
}

  